adagrams/game.py

```diff
@@ -1,11 +1,193 @@
+        'L': 1, 
+        'M': 3, 
+        'N': 1, 
+        'O': 1, 
+        'P': 3, 
+        'Q': 10, 
+        'R': 1, 
+        'S': 1, 
+        'T': 1, 
+        'U': 1, 
+        'V': 4, 
+        'W': 4, 
+        'X': 8, 
+        'Y': 4, 
+        'Z': 10
+    }
+
+
+    for letter in word:
+        if letter.isalpha():
+            result += values_dict[letter.upper()]
+        else:
+            continue
+    if len(word) in extra_point:
+        result += 8
+    return result
+
+
+
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    """
+    Finds highest scoring word and returns a tuple with winning word and score
+    In case of a tie:
+    - 10 letter word wins
+    - if no 10 letter words, shortest word wins
+    - if two words of same length, whichever comes first in the word list wins
+
+    Parameters: a list of strings
+
+    Returns: a tuple with one string and an integer
+    """
+    word_score_list = []
+    highest_scoring_words = []
+    high_score = 0
+
+    # get scores of each word in list, add to new list, and look for highest score
+    for word in word_list:
+        word_score_list.append(score_word(word))
+        if score_word(word) > high_score:
+            high_score = score_word(word)
+    # if only one highest score, return tuple
+    if word_score_list.count(high_score) == 1:
+        high_score_index = word_score_list.index(high_score)
+        return (word_list[high_score_index], high_score,)
+    else:
+        # if tie by score alone, start making list of words with highest score
+        for i in range(len(word_score_list)):
+            if word_score_list[i] == high_score:
+                # if 10 letter word is found, it wins - doesn't matter if there are multiples, b/c 1st one wins
+                if len(word_list[i]) == 10:
+                    return (word_list[i], high_score,)
+                else:
+                    # create list of tie words that are not 10 characters
+                    highest_scoring_words.append(word_list[i])
+        # now look for shortest word - doesn't matter if there are multiples of same length, b/c 1st one wins
+        shortest_word_length = 10
+        shortest_word = ""
+
+        for word in highest_scoring_words:
+            if len(word) < shortest_word_length:
+                shortest_word_length = len(word)
+                shortest_word = word
+        return (shortest_word, high_score,)
+                
+
+    #####OTHER-OPTION########
+    # dict_of_score_words = {}
+    # for word in word_list:
+    #     each_score = score_word(word)
+    #     dict_of_score_words[word] = each_score
+
+    # current_highest_score = dict_of_score_words[word_list[0]]
+    # current_highest_word =  word_list[0]
+    # for word in dict_of_score_words:
+    #     if dict_of_score_words[word] > current_highest_score:
+    #         current_highest_score = dict_of_score_words[word]
+    #         current_highest_word = word
```

Comment:
Since we access the value for `word` in the loop a couple times, we could also create a variable for it here in the for-loop declaration by iterating over `dict_of_score_words.items()`:
```suggestion
    # for word, score in dict_of_score_words.items():
    #     if score > current_highest_score:
    #         current_highest_score = score
    #         current_highest_word = word
```